src/implementations/minecraft_bedrock/server.rs

```diff
@@ -0,0 +1,515 @@
+                })?;
+                *self.process.lock().await = Some(proc);
+                tokio::task::spawn({
+                    use fancy_regex::Regex;
+                    use lazy_static::lazy_static;
+
+                    let event_broadcaster = self.event_broadcaster.clone();
+                    let uuid = self.uuid.clone();
+                    let name = config.name.clone();
+                    let players_manager = self.players_manager.clone();
+                    // let macro_executor = self.macro_executor.clone();
+                    let mut __self = self.clone();
+                    async move {
+                        fn parse_system_msg(msg: &str) -> Option<String> {
+                            lazy_static! {
+                                static ref RE: Regex = Regex::new(r"\[(.*)\]\s(.*)").unwrap();
+                            }
+                            if RE.is_match(msg).ok()? {
+                                RE.captures(msg)
+                                    .ok()?
+                                    .map(|caps| caps.get(2).unwrap().as_str().to_string())
+                            } else {
+                                None
+                            }
+                        }
+                        fn parse_player_joined(system_msg: &str) -> Option<(String, String)> {
+                            lazy_static! {
+                                static ref RE: Regex = Regex::new(r"Player connected:\s*(\w+),\s*xuid:\s*(\d+)").unwrap();
+                            }
+                            if RE.is_match(system_msg).unwrap() {
+                                if let Some(cap) = RE.captures(system_msg).ok()? {
+                                    Some((
+                                        cap.get(1)?.as_str().to_string(),
+                                        cap.get(2)?.as_str().to_string(),
+                                    ))
+                                } else {
+                                    None
+                                }
+                            } else {
+                                None
+                            }
+                        }
+
+                        fn parse_player_left(system_msg: &str) -> Option<String> {
+                            lazy_static! {
+                                static ref RE: Regex = Regex::new(r"Player disconnected:\s*(\w+)").unwrap();
+                            }
+                            if RE.is_match(system_msg).unwrap() {
+                                if let Some(cap) = RE.captures(system_msg).ok()? {
+                                    Some(cap.get(1)?.as_str().to_string())
+                                } else {
+                                    None
+                                }
+                            } else {
+                                None
+                            }
+                        }
```

Comment:
It's best to move these parser function to a new module. I moved the Java ones to a dedicated `line_parser` module, you can check it out on the 0.5.0 branch